exchanges/okcoin/okcoin.go

```diff
@@ -12,613 +12,1656 @@ import (
-func (o *OKCoin) GetAccountWithdrawalFee(ctx context.Context, currency string) ([]GetAccountWithdrawalFeeResponse, error) {
-	requestURL := getWithdrawalFees
-	if currency != "" {
-		requestURL += "?currency=" + currency
+// GetTicker retrieve the latest price snapshot, best bid/ask price, and trading volume in the last 24 hours.
+func (o *Okcoin) GetTicker(ctx context.Context, instrumentID string) (*TickerData, error) {
+	if instrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	var resp []TickerData
+	err := o.SendHTTPRequest(ctx, exchange.RestSpot, getTickerEPL, http.MethodGet, typeMarket, "ticker"+"?instId="+instrumentID, nil, &resp, false)
+	if err != nil {
+		return nil, err
+	} else if len(resp) == 0 {
+		return nil, errors.New("instrument not found")
 	}
+	return &resp[0], nil
+}
 
-	var resp []GetAccountWithdrawalFeeResponse
-	return resp, o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, accountSubsection, requestURL, nil, &resp, true)
+// GetOrderbook retrieve order book of the instrument.
+func (o *Okcoin) GetOrderbook(ctx context.Context, instrumentID string, sideDepth int64) (*GetOrderBookResponse, error) {
+	if instrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	params := url.Values{}
+	params.Set("instId", instrumentID)
+	if sideDepth > 0 {
+		params.Set("sz", strconv.FormatInt(sideDepth, 10))
+	}
+	var resp []GetOrderBookResponse
+	err := o.SendHTTPRequest(ctx, exchange.RestSpot, getOrderbookEPL, http.MethodGet, typeMarket, common.EncodeURLValues("books", params), nil, &resp, false)
+	if err != nil {
+		return nil, err
+	} else if len(resp) == 0 {
+		return nil, fmt.Errorf("%w for instrument %s", errNoOrderbookData, instrumentID)
+	}
+	return &resp[0], nil
 }
 
-// GetAccountWithdrawalHistory retrieves all recent withdrawal records.
-func (o *OKCoin) GetAccountWithdrawalHistory(ctx context.Context, currency string) ([]WithdrawalHistoryResponse, error) {
-	requestURL := getWithdrawalHistory
-	if currency != "" {
-		requestURL += "/" + currency
+// GetCandlesticks retrieve the candlestick charts. This endpoint can retrieve the latest 1,440 data entries. Charts are returned in groups based on the requested bar.
+func (o *Okcoin) GetCandlesticks(ctx context.Context, instrumentID string, interval kline.Interval, after, before time.Time, limit int64) ([]CandlestickData, error) {
+	if instrumentID == "" {
+		return nil, errMissingInstrumentID
 	}
-	var resp []WithdrawalHistoryResponse
-	return resp, o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, accountSubsection, requestURL, nil, &resp, true)
+	params := url.Values{}
+	params.Set("instId", instrumentID)
+	var err error
+	if interval != kline.Interval(0) {
+		var intervalString string
+		intervalString, err = intervalToString(interval, false)
```

Comment:
Please make the `bool` of `false` a param, because wrapper functions `GetHistoricCandles` and `GetHistoricCandlesExtended` need it to be `true` for UTC alignment